.github/workflows/check_for_redirects.yml

```diff
@@ -0,0 +1,51 @@
+name: Redirects Workflow
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize, labeled]
+env:
+  DIFF_DIRECTORIES: 'src/pages src/fragments'
+jobs:
+  onPrOpen:
+    name: Check if redirects are needed on PR opened
+    runs-on: ubuntu-latest
+    if: github.event.action == 'opened'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+      - name: Get count of deleted files
+        run: |
+          git fetch origin main
+          echo "DELETED_FILES_ON_OPENED=$(git diff --name-status --diff-filter=D origin/main -- ${{ env.DIFF_DIRECTORIES }} ${{ github.pull_request.head.sha }} -- ${{ env.DIFF_DIRECTORIES }} | wc -l)" >> $GITHUB_ENV
+          echo "Deleted file count: ${{ env.DELETED_FILES_ON_OPENED }}"
+      - name: Fail status check if there are deleted files
+        if: ${{ env.DELETED_FILES_ON_OPENED > 0 }}
+        run: exit 1
+  onPrSync:
+    name: Check if redirects are needed on PR sync
+    runs-on: ubuntu-latest
+    if: github.event.action == 'synchronize'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+      - name: Get count of deleted files from last sync
+        run: |
+          git fetch origin ${{ github.event.before }} --depth=1
+          echo "DELETED_FILES_ON_SYNC=$(git diff --name-status --diff-filter=D ${{ github.event.before }} -- ${{ env.DIFF_DIRECTORIES }} ${{ github.event.after }} -- ${{ env.DIFF_DIRECTORIES }} | wc -l)" >> $GITHUB_ENV
```

Comment:
For my own knowledge, what is `github.event.after`?

Reply:
It is the SHA of the most recent commit: https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#push

I wanted to get the commit from the latest push and only look at the changes in there